lib/sshkit/ssh/connection.ex

```diff
@@ -34,10 +26,8 @@ defmodule SSHKit.SSH.Connection do
   * `:timeout`: A timeout in ms after which a command is aborted. Defaults to `:infinity`.
   * `:port`: The remote-port to connect to. Defaults to 22.
   * `:user`: The username with which to connect.
-             Defaults to `$LOGNAME`, or `$USER` on UNIX, or `$USERNAME` on windows.
-  * `:password`: The password to login with
-  * `:dry_run`: If set to `true` no actual connection to the remote is established.
-                Instead all commands a logged. Defaults to `false`.
```

Comment:
Hey @tessi, that's right, I removed it for now. The `:ssh_modules`/`:impl` option and the `:dry_run` option were colliding and it wasn't quite clear which should take precedence.

My idea for "dry run" is that it should be handled at a higher level:

We could A) add a dry-run option at a higher level within SSHKit and then pass down an alternate `:impl` or B) I was thinking about maybe doing something much easier:

When running a command via `SSHKit.run`, we could just `echo` it. That would mean, we still connect to the host, change into the correct directory if possible etc. and then also see the result of ENV variable replacement.

This would make the dry-run option much more meaningful I'd say.
I'd be curious to hear your thoughts, maybe we can stick our heads together over lunch?

Reply:
Created issue #100 to keep track…